### PR TITLE
[TypeScript] Lint TypeScipt code

### PR DESCRIPTION
**Pull request by nlordell**
Currently the typescript code is not being linted in the CI. This PR adds some basic recommended typescript linting rules and fixes to code so that the linting succeeds.

In order to fix an import lint on importing `truffle-assertions`, I added some typings for the module.

### Test Plan

CI - linting should succeed.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,34 +1,43 @@
...
     "linebreak-style": ["error", "unix"],
+    "no-console": "error",
     quotes: ["error", "double"],
     semi: ["error", "never"],
-    "prefer-const": ["error"],
-    "no-var": ["error"],
 
-    "no-unused-vars": "off",
-    "@typescript-eslint/no-unused-vars": ["error"],
+    //"prettier/prettier": "error",
```
Comment (e00E) [inline .eslintrc.js]: Should this line be deleted?

**Review by e00E — APPROVED** [review]

**Review by fleupold — APPROVED** [review]
Very cool. It goes without saying that I didn't review all the ts changes, as I assume they were automated.

**Comment by nlordell** [comment]
> Very cool. It goes without saying that I didn't review all the ts changes, as I assume they were automated.

I had to add type return types everywhere. The semi-colon stuff was automated.

I split this out into #737 and #738 which each contain some non-automatic/mechanical changes to appease the linter.

File: migration_conf.js [hunk 2]
```diff
@@ -1,8 +1,6 @@
 const path = require("path")
 
-// eslint-disable-next-line no-undef
 const BUILD_DIR = path.join(__dirname, "build", "contracts")
-// eslint-disable-next-line no-undef
 const NETWORKS_FILE_PATH = path.join(__dirname, "networks.json")
```
Comment (bh2smith) [inline migration_conf.js:4]: Weird... why where these even here. Looks like they were perfectly defined..

File: src/balance_reader.ts [hunk 3]
```diff
@@ -1,8 +1,8 @@
-import BN from "bn.js";
-import Web3 from "web3";
```
Comment (bh2smith) [inline src/balance_reader.ts:2]: Awesome! I love no-semi

**Review by bh2smith — COMMENTED** [review]
I would say that the semis could have been removed separately so I could actually detect what else was changed.

File: migration_conf.js (hunk 2 above)
Comment (nlordell) [reply migration_conf.js:4]: Because the ESLint `env` was incorrectly setup for this file.

**Comment by nlordell** [comment]
> I would say that the semis could have been removed separately so I could actually detect what else was changed.

The issue is that CI would fail as the formatting would be incorrect. I suggest filtering by `.json` and `.js` in the GitHub diff UI.

**Pull request merged**